terraform/get-script-parameters.sh

```diff
@@ -1,17 +1,15 @@
 #!/bin/bash
 
-# Check if number of parameter is correct
-if [[ $# -lt 1 ]] ; then
-    printf 'Error: needs at least 1 parameter(s) - called with %s parameter(s)\n' "$#"
-    printf 'Usage: %s <command ex.init|plan|apply>\n' "$(basename "$0")"
+BASEDIR=$(dirname "$(readlink -f -- "$0")")
+
+# If the user provided a custom set of variable and it doesn't contain the STEP variable,
+# we throw an error because we need it for Terraform scripts
+if [[ -n "$SCRIPT_PARAMS_VARIABLES" && ! "${SCRIPT_PARAMS_VARIABLES[@]}" =~ "STEP" ]]; then
+    printf "Error: STEP variable isn't in your \`SCRIPT_PARAMS_VARIABLES\`. It's needed by Terraform scripts\n" >&2
     exit 1
 fi
 
-# Get the first argument as the terraform step that will be launched
-export STEP=$1; shift
-# Get all other command options
-# We use it to add other arguments to terraform commands
-export COMMAND_OPTIONS="$@"
+source "$BASEDIR"/bash-methods/components/check-bash-parameters.sh
```

Comment:
Hm it won't work, the BASEDIR of the current file doesn't contain `bash-methods` directory

Reply:
You can test it with a repository we use by changing the branch in the `git clone` command in `prepare.sh`

Reply:
I tested it with multiple repos and it works because at the begining of the script there is `BASEDIR=$(dirname "$(readlink -f -- "$0")")` and when you source bash files they are executed in the same shell as the one that sourced the file so the `$0` variable is the same as if we were in the original script